src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs

```diff
@@ -658,8 +658,8 @@ private unsafe void EnsureHeader()
                     _pointerSize = 8;
                     break;
 
-                case (Machine)0x6264: /* LoongArch64 */
-                    _architecture = (Architecture)6; /* LoongArch64 */
```

Comment:
@jkotas @cshung 
The usage of the `Machine.LoongArch64` is ok.
But the `Architecture.LoongArch64` is still wrong.

What should I do ?  keeping the hard coding `(Architecture)6` ?

Reply:
@cshung Does ILCompiler.Reflection.ReadyToRun have to stick to netstandard2.0?

Reply:
We had that discussion before [here](https://github.com/dotnet/runtime/pull/72017). We could upgrade `ILCompiler.Reflection.ReadyToRun` to net6.0, but that doesn't help since the constants aren't available until net7.0.

Reply:
Can we upgrade `ILCompiler.Reflection.ReadyToRun.csproj` to `$(NetCoreAppCurrent)` or `$(NetCoreAppToolCurrent)`? Both of its consumers (dotnet-pgo and R2RDump) are using the current TFM, and so far, `ILCompiler.Reflection.ReadyToRun.Experimental` hasn't been published to nuget.org.

Reply:
I think the problem revolves around usage of the assembly in ILSpy; as you can see, Andrew bumped up the System.Reflection.Metadata version to 7.0 in https://github.com/dotnet/runtime/pull/85738 but even that required a relatively complex roundtrip with the ILSpy team. I think that Andrew's above link points to a discussion thread with more detail.

Reply:
@am11, Yes, the package is not published on nuget.org, it is currently published on the transport feed [here](https://github.com/icsharpcode/ILSpy/blob/4aca5ab09cd22204651383b5c3209738581c1f85/NuGet.config#L6) and consumed [here](https://github.com/icsharpcode/ILSpy/blob/4aca5ab09cd22204651383b5c3209738581c1f85/ILSpy.ReadyToRun/ILSpy.ReadyToRun.csproj#L41).

Reply:
Upgrading ILC.R.R2R project to latest or net7.0 TFM has nothing to do with the fixed dependency version of S.R.M.

Reply:
@cshung Can we delete the Architecture property and return the PE file machine instead? And/or create a local definition of the Architecture property that is local to `ILCompiler.Reflection.ReadyToRun`?

In order to stay independent on the runtime, the parser should not be using enum definitions from the runtime.

Reply:
https://github.com/dotnet/runtime/pull/86277